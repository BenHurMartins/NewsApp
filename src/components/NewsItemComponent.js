import React from 'react';
import {Text, StyleSheet, View} from 'react-native';
import {Colors, Mixins} from '../styles';
import {Divider, ThumbImage, UrlTextComponent} from './';
import {dateFormatter} from '../services/utils';

const NewsItemComponent = ({news}) => {
  const handleClick = () => {
    // dispatch({type: SET_ID, payload: source.id});
    // if (orientation.portrait) navigation.navigate('News');
  };

  return (
    <>
      <View style={styles.container} onPress={handleClick}>
        <View style={styles.titleContainer}>
          <ThumbImage url={news.urlToImage} />
          <Text style={styles.title}>{news.title}</Text>
        </View>
        <View style={styles.descriptionContainer}>
          <Text style={styles.description}>{news.description}</Text>
        </View>
        <UrlTextComponent url={news.url} />
        <View style={styles.authorContainer}>
          <Text style={styles.author}>Author: {news.author}</Text>
        </View>
        <View style={styles.descriptionContainer}>
          <Text style={styles.publishedtime}>
            Published at: {dateFormatter(news.publishedAt)}
          </Text>
        </View>
      </View>
      <Divider style={{alignSelf: 'center', width: Mixins.DEVICE_WIDTH90}} />
    </>
  );
};

export default NewsItemComponent;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'center',
    zIndex: 10,
    padding: 10,
  },
  titleContainer: {
    width: '100%',
    justifyContent: 'flex-start',
    marginBottom: 10,
    flexDirection: 'row',
  },
  title: {
    textAlign: 'justify',
    width: Mixins.DEVICE_WIDTH60,
    fontSize: Mixins.FONT_SIZE_REGULAR,
    fontWeight: 'bold',
    color: Colors.PRIMARY,
  },
  descriptionContainer: {
    width: '90%',
    marginBottom: 10,
  },
  description: {
    fontSize: Mixins.FONT_SIZE_SMALL,
    fontWeight: '200',
    textAlign: 'justify',
  },
  authorContainer: {
    width: '90%',
    marginTop: 10,
  },
  author: {
    fontSize: Mixins.FONT_SIZE_SMALL - 2,
    fontWeight: '800',
    textAlign: 'justify',
    color: Colors.PRIMARY,
  },
  publishedtime: {
    fontSize: Mixins.FONT_SIZE_SMALL - 2,
    fontWeight: '300',
  },
});
