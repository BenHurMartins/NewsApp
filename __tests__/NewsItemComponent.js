import 'react-native';
import React from 'react';
import {render, act} from '@testing-library/react-native';
import NewsItemComponent from '../src/components/NewsItemComponent';
import renderer from 'react-test-renderer';

const news = {
  urlToImage: 'https',
  url: 'https',
  author: 'Ben Hur',
  publishedAt: 1629672740563,
  title: 'test title',
  description: 'Lorem ipsum dolor sit amet',
};

describe('NewsItemComponent component', () => {
  it('NewsItemComponent renders correctly', () => {
    renderer.create(<NewsItemComponent news={news} />);
  });
  it('NewsItemComponent snapshot test', () => {
    const tree = renderer.create(<NewsItemComponent news={news} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should find the author', () => {
    const {getByText} = render(<NewsItemComponent news={news} />);
    act(() => {
      expect(getByText('Author: Ben Hur')).not.toBe(null);
    });
  });
  it('Should find the description', () => {
    const {getByText} = render(<NewsItemComponent news={news} />);
    act(() => {
      expect(getByText('Lorem ipsum dolor sit amet')).not.toBe(null);
    });
  });
  it('Should find the title', () => {
    const {getByText} = render(<NewsItemComponent news={news} />);
    act(() => {
      expect(getByText('test title')).not.toBe(null);
    });
  });
  it('Should find the url', () => {
    const {getByText} = render(<NewsItemComponent news={news} />);
    act(() => {
      expect(getByText('https')).not.toBe(null);
    });
  });
});
