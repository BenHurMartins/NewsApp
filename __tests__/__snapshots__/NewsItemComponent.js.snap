// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItemComponent component NewsItemComponent snapshot test 1`] = `
Array [
  <View
    onPress={[Function]}
    style={
      Object {
        "justifyContent": "center",
        "padding": 10,
        "width": "100%",
        "zIndex": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https",
          }
        }
        style={
          Array [
            Object {
              "borderColor": "#65a1dd",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 187.5,
              "marginRight": 10,
              "resizeMode": "cover",
              "width": 225,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          Object {
            "color": "#65a1dd",
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "justify",
            "width": 450,
          }
        }
      >
        test title
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 22.5,
            "fontWeight": "200",
            "textAlign": "justify",
          }
        }
      >
        Lorem ipsum dolor sit amet
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20.5,
            "fontWeight": "200",
            "textAlign": "justify",
            "textDecorationLine": "underline",
          }
        }
      >
        https
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 10,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#65a1dd",
            "fontSize": 20.5,
            "fontWeight": "800",
            "textAlign": "justify",
          }
        }
      >
        Author: 
        Ben Hur
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20.5,
            "fontWeight": "300",
          }
        }
      >
        Published at: 
        22/08/2021 19:52
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#65a1dd",
          "height": 1,
          "width": 750,
        },
        Object {
          "alignSelf": "center",
          "width": 675,
        },
      ]
    }
  />,
]
`;
